### PR TITLE
BUG: Series.__getitem__ with downstream scalars

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

cc @spencerclark I think this fixes a subset of the issues reported in https://github.com/pydata/xarray/issues/3751, can you confirm?

@jorisvandenbossche IIRC geopandas scalars not being recognized by lib.is_scalar has caused some issues there; does this address any of those?

### CODE REVIEW

File: pandas/core/series.py [hunk 1]
```diff
@@ -840,11 +840,15 @@ def _slice(self, slobj: slice, axis: int = 0) -> "Series":
...
+        key = lib.item_from_zerodim(key)
 
         if key is Ellipsis:
             return self
 
-        key_is_scalar = is_scalar(key)
+        # check for is_list_like/slice instead of is_scalar to allow non-standard
+        #  scalars through, e.g. cftime.datetime needed by xarray
+        #  https://github.com/pydata/xarray/issues/3751
+        key_is_scalar = not is_list_like(key) and not isinstance(key, slice)
```
Comment (jreback) [inline pandas/core/series.py]: this is really really sketch, why don't we just fix is_scalar?

also pls add a test if you can for this

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/series.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/series.py]: > why don't we just fix is_scalar?

Depends on what we want `is_scalar` to mean.  We could re-write `is_scalar` to just match this (actually more performant than what we have now), but it isn't really viable to add checks for any custom scalar that downstream libraries might implement.
 
Going to wait to hear from the downstream folks on if this actually solves their problem(s)

Comment (jreback) [reply pandas/core/series.py]: I actually think is_scalar should just be this its much more generic and future proof

**Comment by spencerkclark** [comment]
Thanks @jbrockmendel -- sorry I just saw this now.  Yes, as written, this change indeed fixes the remaining issue for us, though I think @jreback's suggestion would work too.

File: pandas/core/series.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/series.py]: hmm so changing is_scalar to just match this is breaking a bunch of tests bc `is_scalar(some_lambda_func)` is returning True

Comment (jreback) [reply pandas/core/series.py]: right you might need some more exclusions, e.g. callabes

Comment (jbrockmendel) [reply pandas/core/series.py]: so the trouble im facing ATM is that TimeGrouper is being recognized as a scalar

Comment (jbrockmendel) [reply pandas/core/series.py]: The more I look at this the more skeptical I am of trying to amend is_scalar

Comment (jreback) [reply pandas/core/series.py]: since we don't actually have a test case for this, am inclined to close.

Comment (jbrockmendel) [reply pandas/core/series.py]: coming up with a test case isnt difficult, just need to settle on an approach.

**Comment by jbrockmendel** [comment]
Updated with a better implementation, and a test that fails in master: `df.dtypes[some_dtype]` currently raises, is fixed here.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
much better, thanks.

**Pull request merged**